pyomo/core/base/param.py

```diff
@@ -908,7 +913,16 @@ def __call__(self, exception=True):
         Return the value of this parameter.
         """
         if self._constructed:
-            return _ParamData.__call__(self, exception=exception)
+            if not self._data:
```

Comment:
So an indexed component can be constructed but self._data could be none?  That's surprising.  I think the indexed component documentation should include a discussion of how different construction "modes" initialize the core data structures.